configuration.go

```diff
@@ -172,6 +174,13 @@ func (v *VirtualMachineConfiguration) SetStorageDevicesVirtualMachineConfigurati
 	}
 	array := objc.ConvertToNSMutableArray(ptrs)
 	C.setStorageDevicesVZVirtualMachineConfiguration(objc.Ptr(v), objc.Ptr(array))
+	v.storageDeviceConfiguration = cs
+}
+
+// StorageDevices return the list of storage device configuration configured in this virtual machine configuration.
+// Return an empty array if no storage device configuration is set.
+func (v *VirtualMachineConfiguration) StorageDevices() []StorageDeviceConfiguration {
+	return v.storageDeviceConfiguration
 }
```

Comment:
I took this approach because converting from raw pointers to a slice of interfaces and then to each concrete type is challenging. I plan to apply similar fixes to other similar methods in a separate PR.

Reply:
yeah this was definitely the part I was missing as I didn't change it during my tests, thanks for the fix 🙏